backtracking/minimax.cpp

```diff
@@ -1,27 +1,41 @@
+/**
+ * @file
+ * @brief returns which is the longest/shortest number
+ * using [minimax](https://en.wikipedia.org/wiki/Minimax) algorithm
+ */
+#include <algorithm>
 #include <cmath>
 #include <iostream>
 #include <vector>
 
-using std::cout;
-using std::endl;
-using std::max;
-using std::min;
-using std::vector;
-
-int minimax(int depth, int node_index, bool is_max, vector<int> scores,
-            int height) {
-    if (depth == height)
+/**
```

Comment:
I like the way it is; in `minimax.cpp` there's only one function.
Is it really necessary to create a namespace?

Reply:
Pros of using namespaces:
1. ensures the code is contained in its visibility
2. better documentation
3. adds structure to an otherwise chaotic repo

Cons:
* Bit more work on the author to ensure to write better code
* (anything else?)

Reply:
Sounds good to me, how shall we start? 🙂

Reply:
![image](https://user-images.githubusercontent.com/7001608/87575698-47cd7100-c69e-11ea-947a-f9a237d91daa.png)

Reply:
Done; please check if it is correct.